### PR TITLE
Load multiple model outputs. Interactiveness using ggplotly

### PR DESCRIPTION
**Pull request by shubhamagarwal92**
## Description

1. Change in 'get_workflow_ids' (line 137) db/R/query.dplyr.R allowing a flag to check if we want to load all workflow ids.
2. shiny/workflowPlot/helper.R to allow helper functions. For now, it contains the function to check and download the required packages.
3. shiny/workflowPlot/ui.R allowing multiple selection 
4. shiny/workflowPlot/server.R load multiple model outputs. Interactiveness using ggplotly. Refactored code. 

## Motivation and Context

1. Allows interactiveness to the original plots.
2. Allows loading multiple model outputs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by tonygardella** [comment]
@dlebauer I tagged you as reviewer because you are paired with Shubham this week for code review. @ashiklom I tagged you as reviewer because you are the mentor.

**Comment by tonygardella** [comment]
@shubhamagarwal92 You'll need to do `make document` within your branch and push up the files it creates.

File: shiny/workflowPlot/helper.R [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    } 
+    library(packageName,character.only=TRUE,quietly=TRUE,verbose=FALSE)
+  }
+}
+isInstalled <- function(mypkg){
+  is.element(mypkg, installed.packages()[,1])
+}
+checkAndDownload(c('plotly','scales','dplyr'))
+# devtools::install_github('hadley/ggplot2')
+# testVal = 5
```
Comment (ashiklom) [inline shiny/workflowPlot/helper.R]: This should be removed. Instead, specify package dependencies in the PEcAn.visualization package DESCRIPTION file.

File: shiny/workflowPlot/server.R [hunk 2]
```diff
@@ -3,99 +3,198 @@ library(PEcAn.DB)
...
+  # @return List of variable names
+  var_names_all <- function(workflow_id, run_id){
+    # Get variables for a particular workflow and run id
+    var_names <- get_var_names(bety, workflow_id, run_id)
+    # Remove variables which should not be shown to the user
+    removeVarNames <- c('Year','FracJulianDay')
+    var_names <- var_names[!var_names %in% removeVarNames]
+    return(var_names)
+  }
+  # Update variable names
```
Comment (ashiklom) [inline shiny/workflowPlot/server.R]: These functions should be moved to a package, either `db` or `visualization` (@mdietze I don't remember what we decided, but I think `db`?).

File: shiny/workflowPlot/server.R [hunk 3]
```diff
@@ -3,99 +3,198 @@ library(PEcAn.DB)
...
-        title <- var_name
-        ylab <- ""
-        for(file in files) {
-          nc <- nc_open(file)
+  # Load data for a single run of the model
+  # @param workflow_id and run_id
+  # @return Dataframe for one run 
+  # For a particular combination of workflow and run id, loads
+  # all variables from all files. 
+  load_data_single_run <- function(workflow_id,run_id){
```
Comment (ashiklom) [inline shiny/workflowPlot/server.R]: Move function to package (see above comment).

File: shiny/workflowPlot/server.R (hunk 2 above)
Comment (shubhamagarwal92) [inline shiny/workflowPlot/server.R]: Why do we want this in db package ([get_var_names](https://github.com/PecanProject/pecan/blob/develop/db/R/query.dplyr.R#L183) is already there)? Here I am just subsetting to remove var_names related to 'Year' and 'FracJulianDay' which are not used for plotting.

File: shiny/workflowPlot/server.R (hunk 3 above)
Comment (shubhamagarwal92) [inline shiny/workflowPlot/server.R]: Adding in [query.dplyr.R](https://github.com/PecanProject/pecan/blob/develop/db/R/query.dplyr.R). Is that good?

**Pull request merged**